### PR TITLE
Tod workflow

### PR DESCRIPTION
**Pull request by vshanthe**
## 📝 Description
Integration of tod workflow 

## ✔️ How to Test
Manual testing 
Run make e2e 
once run is completed a report.xml will be generated under hack/
add generation of xml in Makefile - pytest test/integration/ --junitxml="report.xml" 

## 📷 Preview
run link - https://github.com/vshanthe/py-metadata/actions/runs/8053127975/job/21994784871
**If applicable, include a screenshot or code snippet of this change. Otherwise, please remove this section.**

### CODE REVIEW

**Review by jriddle-linode — APPROVED** [review]
LGTM

**Review by lgarber-akamai — APPROVED** [review]

File: hack/run-e2e.yml [hunk 1]
```diff
@@ -129,6 +129,26 @@
             PYTHON={{ venv_python }} make e2e-local
           register: run_tests
 
+        - name: Find report.xml file on remote server
```
Comment (ykim-1) [inline hack/run-e2e.yml:132]: typo here

File: .github/workflows/e2e-suite.yml [hunk 2]
```diff
@@ -43,7 +49,42 @@ jobs:
...
+          status=0
+          if ! make e2e > test_output.txt; then
+            echo "EXIT_STATUS=1" >> $GITHUB_ENV
+          fi
+            cat test_output.txt
         env:
           CLEANUP_TEST_LINODE_INSTANCE: true
           LINODE_TOKEN: ${{ secrets.DX_LINODE_TOKEN }}
+          
+      - name: Copy report.xml from hack directory to root directory
```
Comment (ykim-1) [inline .github/workflows/e2e-suite.yml:62]: Can we skip this step?

**Review by ykim-1 — APPROVED** [review]
Looks good besides some minor fixes.

File: .github/workflows/e2e-suite.yml (hunk 2 above)
Comment (vshanthe) [reply .github/workflows/e2e-suite.yml:62]: i tried with ur suggestion to have hack/filename but it's getting failed

**Pull request merged**